go/libkb/naclwrap.go

```diff
@@ -700,20 +707,32 @@ func (k NaclDHKeyPair) EncryptToString(plaintext []byte, sender GenericKey) (str
 	return PacketArmoredEncode(info)
 }
 
-func (k NaclDHKeyPair) SecretSymmetricKey(reason EncryptionReason) ([]byte, error) {
-
+func (k NaclDHKeyPair) SecretSymmetricKey(reason EncryptionReason) (NaclSecretBoxKey, error) {
 	if !k.CanDecrypt() {
-		return nil, NoSecretKeyError{}
+		return NaclSecretBoxKey{}, NoSecretKeyError{}
 	}
-	if len(reason) < 8 {
-		return nil, KeyGenError{Msg: "reason must be at least 8 bytes"}
+
+	return DeriveSymmetricKey([32]byte(*k.Private), reason)
+}
+
+func DeriveSymmetricKey(inKey NaclSecretBoxKey, reason EncryptionReason) (NaclSecretBoxKey, error) {
```

Comment:
I like sharing the key derivation logic with chat local storage. Though this assumes that we never use KBFS keys + HMAC as an actual public authenticator. That sounds like...the kind of thing we might do in some random part of the KBFS protocol? Maybe someday in the future? :(

Using the key as hash/HMAC _input_ feels less likely to collide with some other plausibly-safe-to-publish use of the key. What do you think?